BoxContentSDK/BoxContentSDK/Operations/BOXAPIOperation.m

```diff
@@ -540,13 +540,15 @@ - (void)processResponse:(NSURLResponse *)response
 
 - (void)finishURLSessionTaskWithData:(NSData *)data response:(NSURLResponse *)response error:(NSError *)error
 {
-    if (self.HTTPResponse == nil) {
+    if (self.HTTPResponse == nil && response != nil) {
         // If the response has not already been handled (some, but not all, operations get
-        // 'processResponse:' earlier in the process) handle it here.
+        // 'processResponse:' earlier in the process), handle it here.
+        // Also, do not handle a nil response. That could happen if there was a failure on the whole
+        // network connection, for example. In that case, there is no response.
         [self processResponse:response];
     }
 
-    if (data != nil) {
+    if ([data length] > 0) {
```

Comment:
could you add a comment why we have to check for non-empty data?

would there be a legitimate case that data is empty and needs to be processed accordingly inside processResponseData, e.g. to set error accordingly?